### PR TITLE
Add AppContainer and move bootstrapping to App component

### PR DESCRIPTION
**Pull request by jakemcdermott**
**components/AppContainer**
- wraps patternfly and awx components only. Encapsulate the UI that does not vary across navigable screens.
- is not concerned with current router implementation or routing
- is not concerned with current choice of i18n lib or injecting i18n dependencies
- room for approval notifications and other stuff we'll be adding soon that need to live in the common shell that surrounds navigable screens.

**App**
- now contains the typical app bootstrapping stuff
- knows about our choice of router and how it is implemented
- knows about our choice of i18n lib and how it is configured

**index**
- now just the typical react entry point for the whole program. no extra stuff.

### CODE REVIEW

File: awx/ui_next/src/components/AppContainer/AppContainer.jsx [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+      color: inherit;
+    }
+
+    & svg {
+      height: 76px;
+    }
+  }
+`;
+
+class AppContainer extends Component {
```
Comment (jakemcdermott) [inline awx/ui_next/src/components/AppContainer/AppContainer.jsx]: I'm converting this to a functional component but haven't pushed the changes to this PR yet.

File: awx/ui_next/src/App.jsx [hunk 2]
```diff
@@ -1,216 +1,89 @@
...
+import Login from './screens/Login';
+
+import ja from './locales/ja/messages';
+import en from './locales/en/messages';
+import { isAuthenticated } from './util/auth';
+import { getLanguageWithoutRegionCode } from './util/language';
+
+import getRouteConfig from './routeConfig';
+
+const AppRoute = ({ auth, children, ...rest }) =>
```
Comment (AlexSCorey) [inline awx/ui_next/src/App.jsx]: This function is confusing, but you and I were able to clear it up offline.

**Review by AlexSCorey — CHANGES_REQUESTED** [review]
Everything looks good to me except that nested ternary operator function.  We spoke offline and came up with a plan. Ping me when this PR is updated.

File: awx/ui_next/src/App.jsx [hunk 3]
```diff
@@ -1,216 +1,89 @@
...
+import Login from './screens/Login';
+
+import ja from './locales/ja/messages';
+import en from './locales/en/messages';
+import { isAuthenticated } from './util/auth';
+import { getLanguageWithoutRegionCode } from './util/language';
+
+import getRouteConfig from './routeConfig';
+
+const ProtectedRoute = ({ children, ...rest }) =>
```
Comment (AlexSCorey) [inline awx/ui_next/src/App.jsx:24]: Nice!

File: awx/ui_next/src/App.jsx [hunk 4]
```diff
@@ -1,216 +1,89 @@
...
+  isAuthenticated(document.cookie) ? (
+    <Route {...rest}>{children}</Route>
+  ) : (
+    <Redirect to="/login" />
+  );
+
+const UnprotectedRoute = ({ children, ...rest }) =>
```
Comment (AlexSCorey) [inline awx/ui_next/src/App.jsx]: Nice

File: awx/ui_next/src/App.jsx (hunk 2 above)
Comment (jakemcdermott) [reply awx/ui_next/src/App.jsx]: You're right! Should be fixed now.

**Review by AlexSCorey — APPROVED** [review]
Nice! Good work.

**Pull request merged**